torchci/scripts/check_alerts.py

```diff
@@ -306,15 +356,18 @@ def main():
     (jobs_to_alert_on, flaky_jobs) = classify_jobs(job_names, sha_grid)
 
     # Fetch alerts
-    existing_alerts = fetch_alerts()
+    existing_alerts = fetch_alerts(TEST_INFRA_REPO_NAME, PYTORCH_ALERT_LABEL)
+    existing_no_flaky_tests_alerts = fetch_alerts(
+        TEST_INFRA_REPO_NAME, NO_FLAKY_TESTS_LABEL
+    )
 
     # Alerts should also be cleared if the current status of HUD is green
-    if len(jobs_to_alert_on) == 0 :
+    if len(jobs_to_alert_on) == 0:
         print("Didn't find anything to alert on.")
 
         if trunk_is_green(sha_grid):
             clear_alerts(existing_alerts)
-            existing_alerts = [] # all alerts have now been cleared
+            existing_alerts = []  # all alerts have now been cleared
```

Comment:
I haven't thought about this deeply yet, but I suspect the alert clearing rule should be different for the no flaky test detector

Reply:
I think that is currently the case. I didn't touch this part. Do you mean that the flaky test alerts should be cleared if trunk is green and the alert doesn't hold any more? In that case flaky test alert should only be cleared if we detect flaky tests (i.e. the bot is working).